include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -471,6 +471,88 @@ def NVMMASharedEncodingAttr :
   let hasCustomAssemblyFormat = 1;
 }
 
+def SwizzledBlocksSharedEncodingAttr :
+  TritonGPU_Attr<"SwizzledBlocksSharedEncoding", "swizzled_blocks_shared_encoding", [SharedEncodingTrait, LayoutEncodingTrait]> {
+  let mnemonic = "swizzled_blocks_shared";
+
+  let description = [{
+This shared encoding is similar to SwizzledSharedEncodingAttr,
+but instead of repeating swizzling pattern every `blockSize = maxPhase*perPhase` rows of memory object,
+this layout changes swizzling pattern.
+
+Swizzling pattern affects only 2 fastest dimensions of a tensor:
+- fastest dimension is called row
+- second fastest dimension is called column
+
+Example below follow this notation, i.e. elemens in a row stored in memory consequentially.
+
+Let's consider an element with position = (rowId, colId).
+For simplicity, we do not vectorize memory in examples,
+i.e. vec == 1 and layout swizzles inidividual elements.
+For vec != 1 example, take a look at SwizzledSharedEncodingAttr documentation.
+
+Swizzled coordinates within memory object are (rowId, swizzledColId):
```

Comment:
(rowId, colId) refers to the coordinates of the element in the logical tensor.
Then for the element in shared memory, the 2D indices should be `(new_rowId, swizzledColId)`, where `new_rowId = rowId * colSizeInBytes / (#banks * 4)`.